rust/parquet/src/util/test_common/file_util.rs

```diff
@@ -19,17 +19,8 @@ use std::{env, fs, io::Write, path::PathBuf, str::FromStr};
 
 /// Returns path to the test parquet file in 'data' directory
 pub fn get_test_path(file_name: &str) -> PathBuf {
-    let mut pathbuf = match env::var("PARQUET_TEST_DATA") {
-        Ok(path) => PathBuf::from_str(path.as_str()).unwrap(),
-        Err(_) => {
-            let mut pathbuf = env::current_dir().unwrap();
-            pathbuf.pop();
-            pathbuf.pop();
-            pathbuf
-                .push(PathBuf::from_str("cpp/submodules/parquet-testing/data").unwrap());
-            pathbuf
-        }
-    };
+    let mut pathbuf =
+        PathBuf::from_str(&arrow::util::test_util::parquet_test_data()).unwrap();
```

Comment:
Yea, better to move the parquet test data fn into this crate, because `arrow` is an optional dependency

Reply:
Agree with @nevi-me

Actually I had asked similar question "where to put them?", finally I decided leaving this question to reviewers, because the most important thing at that time is coding.

At present, with this PR, thanks to @alamb, we finally got a chance to discuss this issue. Actually I don't mind how to improve or split the code in #8967 at all, as long as the fundamental purpose could be achieved, and they do not mess up the existing code  too much :)

Reply:
Will do

Reply:
@nevi-me  and @mqy  -- I tried to move `parquet_test_data` into https://github.com/apache/arrow/blob/master/rust/parquet/src/util/test_common/file_util.rs -- however, the code quickly got messy because `parquet::util` is not publically exported and thus I can't use functions defined there in places (like datafusion) outside the parquet crate. Furthermore, the `test_utils` are only compiled in `test` config, but several datafusion examples use the parquet test data but they are not compiled in `test` config. 

I can think of several possibilities:
1. Leave the `parquet_test_data` function in the arrow crate as it is in this PR
2. Make a copy of parquet_test_data in the parquet crate
3. Make the parquet util module public and export test_util in all configurations

Given that this function is used in tests and the other options seem messy to me, I suggest number 1 (though perhaps I am being lazy)

Reply:
+1 for the first choice right now. Perhaps these functions should stay in a top crate parallel to arrow crate, as common utilities for testing or something else. We may crate that crate later.